eip/EIPS/eip-4337.md

```diff
@@ -317,6 +324,28 @@ Exceptions to the forbidden opcodes:
    (that is, making calls is allowed, using `gasleft()` or `gas` opcode directly is forbidden)
 
 ### Reputation scoring and throttling/banning for paymasters
+#### Rationale
+
+Paymasters might influence many UserOperations in the mempool.
+To prevent abuse, we "penalize" a paymaster that causes invalidation of large number of UserOperations in the mempool.
+To prevent paymasters from "sybil-attack", we require a paymaster to stake with the system, and thus make such DoS attack very expensive.
+Note that this stake is never slashed, and can be withdrawn any time (after unstake delay)
+
+The stake value is not enforced on-chain, but specifically by each node while simulating a transaction.
+The stake is expected to be above PAYMASTER_MIN_STAKE_VALUE, and unstake delay above PAYMASTER_MIN_UNSTAKE_DELAY
+The value of PAYMASTER_MIN_UNSTAKE_DELAY is 84600 (one day)
+
+The value of PAYMASTER_MIN_STAKE_VALUE is determined per chain, as roughly 1000 USD:
```

Comment:
do you have a better suggestion?
we do need to specify some values. at first, I put a $1000 value, saying you need "equivalent". But this force the bundler to contact some exchange to keep it up to date, which is a burden we don't want.
Worse, it means a valid permissionless paymaster today might become under-staked in the future. So instead we thought we can use fixed native values, and only change them on huge price changes.

Reply:
I'm also not a fan of setting it to a fixed USD denominated value.  I think it should be a network-wide value (or a mempool-wide value since a chain could have more than one mempool), which reflects the value and use cases of that network.  E.g. a gaming chain where censorship and frontrunning is not that much of an issue might select a lower value.

Maybe the ERC itself is not the right place to set the values for the different networks, but we do need one place that reflects the consensus for each network.  My current thinking is that the bundler specs and compliance tests suite is the right place for this, since all bundlers should (hopefully) run these tests to ensure compatibility.

If it's in the bundler specs/tests, anyone is free to send a PR to change it, but hopefully it'll involve some process in each network, where their community (or owner - in some networks) decides on it.

The question is what to write about it here in the ERC.  "A chain-specific amount agreed by the mempool nodes, sufficient to mitigate DoS and censorship attacks?"  - the problem is that determining a sufficient amount requires understanding the attack and its economics, which is a bit long to add here.

Reply:
removed actual values. still we need to find a place to communicate them.

Reply:
Yes.  In the bundler specs, reference bundler, and bundlers test suite.  Any bundler that passes these tests can be used by paymaster developers to ensure that the paymaster is sufficiently staked.

Bundlers should also communicate it to users/devs somehow.  I suggest a `getMinimumPaymasterStake(chainid)` RPC which returns the minimum stake value and unstake time for that chain.